alarm_base/src/frame.rs

```diff
@@ -1,16 +1,17 @@
 //! Base types for page frame allocators.
 use alloc::allocator::AllocErr;
+use hal9000::mem::Page;
 
 /// An allocator that provides page frames.
 pub unsafe trait Allocator {
     /// Architecture-dependent size of a physical page.
-    const FRAME_SIZE: usize;
+    const FRAME_SIZE: usize = <Self::Frame as Page>::SIZE;
```

Comment:
Yeah, that's right. In this case it's necessary to do that because the `Frame` type might implement any number of traits that might define an associated constant `SIZE`, so we have to use this syntax to disambiguate it.

This [blog post](http://smallcultfollowing.com/babysteps/blog/2016/11/02/associated-type-constructors-part-1-basic-concepts-and-introduction/#background-traits-and-associated-types) might be helpful!